docs/spec/dataclasses.rst

```diff
@@ -526,6 +532,56 @@ This includes, but is not limited to, the following semantics:
   dataclass must be explicitly annotated as e.g. ``x: ClassVar[Final[int]] =
   3``.
 
+Converters
+^^^^^^^^^^
+
+The ``converter`` parameter can be specified in a field definition to provide a single-argument callable
+used to convert values when assigning to the associated attribute. This feature allows for automatic type
+conversion and validation during object initialization and attribute assignment.
+
+Converter behavior:
+
+* For non-frozen dataclasses, the converter is used for all attribute assignment, including assignment
```

Comment:
I find it a little confusing for this section to refer to "non-frozen dataclasses," since that sounds to me like you're talking about stdlib dataclasses. But stdlib dataclasses don't have the `converter` parameter, right?

Maybe you could write this as "non-frozen classes impacted by `dataclass_transform`" and then just "frozen classes" or "non-frozen classes" in the rest of this section.